src/main/java/gyro/google/iam/ServiceAccountResource.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.iam;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.api.services.iam.v1.Iam;
+import com.google.api.services.iam.v1.model.CreateServiceAccountRequest;
+import com.google.api.services.iam.v1.model.DisableServiceAccountRequest;
+import com.google.api.services.iam.v1.model.EnableServiceAccountRequest;
+import com.google.api.services.iam.v1.model.ServiceAccount;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+import gyro.google.GoogleResource;
+import gyro.google.util.Utils;
+
+/**
+ * Creates a service account.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *      google::service-account example-service-account
+ *          name: "example-service-account"
+ *          display-name: "example-service-account"
+ *          description: "example service account"
+ *      end
+ */
+@Type("service-account")
+public class ServiceAccountResource extends GoogleResource implements Copyable<ServiceAccount> {
+
+    private String displayName;
+    private String description;
+    private String name;
+    private Boolean enableAccount;
+
+    // Read-only
+    private String id;
+    private String email;
```

Comment:
How is this an output field, where does it get the email from ?

Reply:
The email gets set in the copyFrom. It's a part of the id and it uniquely identifies an account.